Org Protocol Handler.app/Contents/Resources/parse.py

```diff
@@ -1,25 +1,38 @@
 #!/usr/bin/env python
 
+import ConfigParser
+import os
+import subprocess
 import sys
-import urlparse
 import urllib
-import subprocess
-import os
-import ConfigParser
+import urlparse
 
 
-def get_emacsclient_path():
-    """Get the configured path to `emacsclient`, or the default."""
+def read_config():
     ini_path = os.path.expanduser("~/.orgprotocol.ini")
-    if os.path.exists(ini_path):
-        config = ConfigParser.ConfigParser()
-        try:
-            config.read([ini_path])
-            return config.get("emacsclient", "path")
-        except Exception, e:
-            pass
+    config = ConfigParser.ConfigParser()
+    config.read([ini_path])
```

Comment:
Realizing that I didn't actually *handle* the exception, the reason for the `try` here originally is that failures within this script will happen invisibly since this is run from within the AppleScript. This particular statement can fail if the config file doesn't exist or its contents can't be parsed for whatever reason.

I think it would make sense to emit a log of some kind. I had put off investigating the "right" way to do this in macOS, but at least spitting out a log file with a stack trace in it would be a huge help to the end user.

Up to you whether you want to try to take that on as part of this patch, or, if not, maybe file an issue so I remember to do it later?

Reply:
Maybe it would be better to just have a simple test suite that users can run.  Running the tests could provide necessary debug information for when this doesn't work out of the box.

Reply:
I removed the `if os.path.exists` check because `ConfigParser` returns an empty list if the ini file doesn't exist.  Since the `config.get` calls are in try/except blocks, that shouldn't be an issue.  In that case `cmd` should end up being the default path to `emacsclient`.